packages/contentstack-migration/examples/05-Update-reference-entry-from-mapper.js

```diff
@@ -0,0 +1,138 @@
+const fs = require('fs');
+const chalk = require('chalk');
+module.exports = async ({ migration, stackSDKInstance, managementAPIClient, config }) => {
+  const modules = ['entries', 'assets', 'extensions'];
+
+  const readAllModulesUids = (filePath) => {
+    let uidMapping = [];
+    modules.forEach((module) => {
+      if (fs.existsSync(`${filePath}/mapper/${module}/uid-mapping.json`)) {
+        let mappedIds = fs.readFileSync(`${filePath}/mapper/${module}/uid-mapping.json`, 'utf-8');
+        mappedIds = JSON.parse(mappedIds);
+        uidMapping = { ...uidMapping, ...mappedIds };
+      } else {
+        console.log(chalk.red(`Mapper Does not exist for module '${module}'`));
+      }
+    });
+
+    return uidMapping;
+  };
+
+  const getEntries = async (ct) => {
+    try {
+      let entries = [];
+      let skip = 0;
+      let limit = 1;
+      let count = 0;
+      while (skip <= count) {
+        const res = await stackSDKInstance.contentType(ct).entry().query({ include_count: true, skip, limit }).find();
+        count = res.count;
+        skip += limit;
+        entries = [...entries, ...res.items];
+      }
+      return entries;
+    } catch (err) {
+      console.log(err);
+      console.log(chalk.red(`No entry is found for Content-type ${ct}`));
+      return [];
+    }
+  };
+
+  const getContentTypeSchema = async (ct) => {
+    try {
+      return await stackSDKInstance.contentType(ct).fetch();
+    } catch (err) {
+      console.log(chalk.red(`Content-type '${ct}' is not present in stack`));
+      return {};
+    }
+  };
+
+  const updateEntryTask = () => {
+    return {
+      title: 'Update Reference Entries from Mapper',
+      successMessage: 'Reference Updated Successfully',
+      failedMessage: 'Failed to Update References in Entries',
+      task: async (params) => {
+        try {
+          const log = console.log;
+
+          if ((!config.contentTypes && !Array.isArray(config.contentTypes)) || !config['mapper-path']) {
+            throw Error(
+              `Content-type or the Mapper Path is missing from the config. Please provide Content-type in Array and mapper path`,
+            );
+          }
+
+          for (let ct of config.contentTypes) {
+            let { schema: ctSchema } = await getContentTypeSchema(ct);
+            // console.log(ctSchema);
+            if (!ctSchema) {
+              continue;
+            }
+
+            let keys = ctSchema?.map((schema) => {
+              return schema.uid;
+            });
+
+            let entry = await getEntries(ct);
+
+            if (entry.length === 0) {
+              log(chalk.red(`No entry found for the CT Content-type '${ct}'`));
+              continue;
+            }
+            let uidMapping = readAllModulesUids(config['mapper-path']);
+
+            for (let e of entry) {
```

Comment:
@cs-raj [Minor] We can divide the logic into several functions rather than writing it all in one. so that debugging and fixing the issues would be simple and also it will increase readability.

Reply:
Yes divided it into 3 different functions